.github/workflows/github-release.yml

```
@@ -0,0 +1,37 @@
+name: github-release
+on:
+  push:
+    tags:
+    - "v[0-9]+.[0-9]+.[0-9]+"
```

We don't really publish pre-releases often, but I wonder if it'll be better to use the `if: startsWith(github.ref, 'refs/tags/')` that's mentioned in the gh-release action instead?

---

It might be easier to just do it in ci yeah, but I do see benefits to doing it in it's own file/action as some of our projects won't use this action at all.

---

Oh, I see, this piece is required otherwise we'll have to run on every push or at best to every push to the main branch.
Maybe we can add a second match for builds something like
```suggestion
    # match semver versions
    - "v[0-9]+.[0-9]+.[0-9]+"
    # match semver pre-releases
    - "v[0-9]+.[0-9]+.[0-9]+-*?"
```